packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -743,9 +743,10 @@ export default Adapter.extend(BuildURLMixin, {
     @param  {Number} status
     @param  {Object} headers
     @param  {Object} payload
+    @param  {String} errorMessage
     @return {Object | DS.AdapterError} response
   */
-  handleResponse: function(status, headers, payload) {
+  handleResponse: function(status, headers, payload, errorMessage) {
```

Comment:
I feel like we should not call the param `errorMessage`. It can be `ok`, or `created` in case of success. I would be ok with `statusText`, even if in case of obvious errors we "augment" the proper `statusText`.